### PR TITLE
Added filter to points in Lines, adapted algorithms according to changes, standardized outputs to return frame and angle

### LINKED ISSUE #11: Standardize outputs for Scanning Algorithm

### PR DESCRIPTION
**Pull request by abhigyandabla**

### CODE REVIEW

File: algorithms/utils/Lines.py [hunk 1]
```diff
@@ -53,7 +53,8 @@ def getIntersections(lines, min_slope=1):
         points: [x]
     """
     intersections = []
-    points = []
+    # xPoints = []
+    # yPoints = []
```
Comment (L0Lmaker) [inline algorithms/utils/Lines.py]: Delete?

File: algorithms/utils/Lines.py [hunk 2]
```diff
@@ -97,9 +98,11 @@ def getIntersections(lines, min_slope=1):
                 # the intersections array is an array of points coordinates (x, y)
                 # the points array is an array of the x coordinates of the points
                 intersections.append(intersect)
-                points.append(intersect[0])
+                # xPoints.append(intersect[0])
+                # yPoints.append(intersect[1])
```
Comment (L0Lmaker) [inline algorithms/utils/Lines.py]: same

File: algorithms/utils/Lines.py [hunk 3]
```diff
@@ -185,20 +188,37 @@ def drawLinesOnFrame(lines, frame):
...
+def filterPoints(numArray):
+    low = np.percentile(numArray, 20)
+    high = np.percentile(numArray, 80)
+    filtered = []
+    
+    for num in numArray:
+        if num >= low and num <= high:
+            filtered.append(num)
+        
+    return filtered
```
Comment (L0Lmaker) [inline algorithms/utils/Lines.py]: documentation

File: algorithms/utils/Lines.py [hunk 4]
```diff
@@ -185,20 +188,37 @@ def drawLinesOnFrame(lines, frame):
...
         else:
-            IntersectingX = np.mean(points)
+            IntersectingX = np.mean(filteredX)
+            IntersectingY = np.mean(filteredY)
 
-        cv2.circle(frame, (int(IntersectingX), int(frame.shape[1] / 2)), 8, (255, 0, 0), -1)
+        cv2.circle(frame, (int(IntersectingX), int(IntersectingY)), 8, (255, 0, 0), -1)
 
-        return (int(IntersectingX), int(frame.shape[1] / 2))
+        return (int(IntersectingX), int(IntersectingY))
```
Comment (L0Lmaker) [inline algorithms/utils/Lines.py]: update documentation for this change and any other methods that are changed

File: algorithms/CenterRowAlgorithm.py [hunk 5]
```diff
@@ -65,14 +66,24 @@ def processFrame(self, frame, show=True):
         cv.fillPoly(black_frame, pts=contours, color=self.contour_color)
         lines, slopes, ellipse_frame = self.ellipse_slopes(contours, black_frame)
         Lines.drawLinesOnFrame(lines, black_frame)
-        intersections, points = Lines.getIntersections(lines)
-        vanishing_point = Lines.drawVanishingPoint(ellipse_frame, points)
+
+        intersections = Lines.getIntersections(lines)
+        xPoints = [point[0] for point in intersections]
+        yPoints = [point[1] for point in intersections]
+        vanishing_point = Lines.drawVanishingPoint(ellipse_frame, xPoints, yPoints)
```
Comment (L0Lmaker) [inline algorithms/CenterRowAlgorithm.py]: python variable conventions, here and everywhere else.

File: algorithms/MiniContoursAlgorithm.py [hunk 6]
```diff
@@ -187,12 +190,12 @@ def getCenterHoughLines(self, frame, num_strips=60, lines_max=30, threshold=4, m
...
-        cv2.imshow('c_mask', c_mask)
-        cv2.imshow('points', points)
-        # cv2.waitKey(1)
-
+        if show:
+            cv2.imshow('frame', frame)
+            cv2.imshow('mask', mask)
+            cv2.imshow('c_mask', c_mask)
+            cv2.imshow('points', points)
+            cv2.waitKey(1)
```
Comment (L0Lmaker) [inline algorithms/MiniContoursAlgorithm.py:198]: See if we can remove unnecassary frame processes for algorithms

**Review by L0Lmaker — CHANGES_REQUESTED** [review]

**Comment by AnimeAllstar** [comment]
When will this be merged by? If this successfully standardizes the outputs, can we merge and move the other changes requested by @L0Lmaker to another PR?

**Comment by L0Lmaker** [comment]
Yeah that should be doable @AnimeAllstar

I'll merge it in and ping you.

**Comment by L0Lmaker** [comment]
> When will this be merged by? If this successfully standardizes the outputs, can we merge and move the other changes requested by @L0Lmaker to another PR?

So it looks like we can't merge this in yet. There are runtime errors when the code is run on videos.

**Pull request closed without merging**